### PR TITLE
Strip container builds of source files etc. 

### LINKED ISSUE #68: Dockerfiles: don't include tool source codes in container builds
**Describe the issue**
The way the dockerfiles are now set up will include the source codes and other unused files in the built container. 
Using `gwama` in `gwas.sif` as an example (around https://github.com/comorment/containers/blob/748a23312ade04a7e397e06fc0a779e192875242/src/dockerfiles/gwas/Dockerfile#L107), a working directory `/tools/gwama/` is created, the installer script is moved to `/tmp/` (good!) but the routine
```
WORKDIR /tools/gwama
COPY /scripts/install_gwama.sh /tmp
RUN chmod +x /tmp/install_gwama.sh
RUN bash /tmp/install_gwama.sh
```
still result in leftover files (.cpp, .h, .zip, binary files) in the working directory, while the binary `GWAMA` is appropriately copied to `/usr/bin/`. 

Unless I miss something, including such files in the containers provided here seems redundant, and inflate their size:
```
$ singularity run singularity/gwas.sif du -h -d 1 /tools/ 
725K    /tools/.waf3-2.0.19-597b128bf9f9d1a665be0bd3cba5812c
109M    /tools/3rd_party
9.2M    /tools/Minimac4
26K     /tools/R
71K     /tools/appcontext
58K     /tools/apps
44M     /tools/bcftools
512     /tools/bgen.tgz
357M    /tools/bolt
15M     /tools/build
35K     /tools/db
296K    /tools/doc
15M     /tools/example
4.2M    /tools/flashpca
47M     /tools/gcta
4.7M    /tools/gctb
80K     /tools/genfile
590K    /tools/gwama
54M     /tools/htslib
3.1M    /tools/king
2.0M    /tools/metal
40M     /tools/plink
41M     /tools/plink2
41M     /tools/plink2_avx2
54M     /tools/prsice
/usr/bin/du: cannot read directory '/tools/qctools/qctool_v2.0.6-Ubuntu16.04-x86_64': Permission denied
137M    /tools/qctools
21M     /tools/regenie
9.9M    /tools/simu
34K     /tools/src
403K    /tools/test
99M     /tools/vcftools
1.1G    /tools/ 
```

### PR DESCRIPTION
**Pull request by espenhgn**
Fixes #68

Changes proposed in this pull request:

- Shrinks the `gwas.sif` build from 861MB to 387MB; using `/tools/` only for BOLT-LMM binary and linked library.
- Using `/tmp/` dir during Docker builds (`gwas.sif`, `r.sif`, `python3.sif`, `hello.sif`) 
- Added license information column in the software table in [src/README.md](https://github.com/comorment/containers/blob/main/src/README.md)

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/comorment/containers/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/comorment/containers/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: src/scripts/install_bolt.sh [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+export VERSION="v2.3.5"
+                           
+wget --no-check-certificate https://storage.googleapis.com/broad-alkesgroup-public/BOLT-LMM/downloads/old/BOLT-LMM_$VERSION.tar.gz
+tar -xvzf BOLT-LMM_$VERSION.tar.gz
+rm -rf BOLT-LMM_$VERSION.tar.gz
+mv BOLT-LMM_$VERSION/* .
+
+# delete some stuff that should not be containerized:
+rm -rf BOLT-LMM_$VERSION
+rm license.txt
```
Comment (ofrei) [inline src/scripts/install_bolt.sh:12]: Is it fine to ``rm license.txt``? This is probably a standard copy of the GPLv3, which mainly governs source code distribution, but not so much the binary distribution (at least that is as far as I can understand all of these Open Source licensing stuff). Yet to be safe I'd probably keep the license - though I also don't feel strongly against removing it.

**Review by ofrei — APPROVED** [review]
Great! I agree with changes, and will follow the same pattern (remove temporary stuff from containers) from now on.

File: src/scripts/install_bolt.sh (hunk 1 above)
Comment (espenhgn) [reply src/scripts/install_bolt.sh:12]: It's GPLv3 in this case plus MIT for NLopt which is included in the build. 
Perhaps we can document the license information for each included software in the `src/README` instead?

Comment (espenhgn) [reply src/scripts/install_bolt.sh:12]: Btw, I messed around a bit with building BOLT-LMM using system libraries (libboost, etc.), but the resulting container size got pretty big (>1GB), so probably not something we should do unless someone reports a problem.

**Pull request merged**